c_src/otter_nif.cpp

```diff
@@ -765,16 +765,16 @@ static ERL_NIF_TERM otter_invoke(ErlNifEnv *env, int argc,
           }
           type_index_resindex[args[i]][i] = value_slot;
         } else {
-          auto wrapper_it = struct_type_wrapper_registry.find(p.type);
-          if (wrapper_it != struct_type_wrapper_registry.end()) {
-            args[i] = &wrapper_it->second.ffi_struct_type;
+          auto wrapper_it = FFIStructTypeWrapper::create_from_tuple(env, p.type_term);
+          if (wrapper_it != nullptr) {
+            args[i] = &wrapper_it->ffi_struct_type;
             void *resource_obj_ptr = nullptr;
             const bool resource = enif_get_resource(
-              env, p.term, wrapper_it->second.resource_type,
+              env, p.term, wrapper_it->resource_type,
               &resource_obj_ptr);
             if (!resource) {
               ready = 0;
-              error_msg = "failed to get resource for struct: " + wrapper_it->second.struct_id;
+              error_msg = "failed to get resource for struct: " + wrapper_it->struct_id;
```

Comment:
will `wrapper_it` point to a nullptr here?will wrapper_it points to a nullptr here?